serde/src/de/mod.rs

```diff
@@ -1866,6 +1885,24 @@ pub trait MapAccess<'de> {
         self.next_value_seed(PhantomData)
     }
 
+    /// Equivalent to calling `Error::duplicate_field` if the field has already
+    /// been deserialized, and to calling `next_value` otherwise.
+    ///
+    /// This method exists purely to reduce the size of the generated code,
+    /// reducing compile times. `MapAccess` implementations should not override
+    /// the default behavior.
+    #[doc(hidden)]
+    #[inline]
+    fn next_value_checked<V>(&mut self, field: &Option<V>, field_name: &'static str) -> Result<V, Self::Error>
+    where
+        V: Deserialize<'de>,
+    {
+	if field.is_some() {
```

Comment:
Drive-by nit: the indentation seems broken here.

Reply:
Good catch, I will fix it.